modules/class-simplechart-plugin-versions.php

```diff
@@ -0,0 +1,145 @@
+<?php
+
+/*
+ * This class manages when we show alerts to upgrade
+ */
+
+class Simplechart_Plugin_Versions {
+
+	private $_simplechart_releases_url = 'https://api.github.com/repos/alleyinteractive/wordpress-simplechart/releases';
+
+	private $_simplechart_url = 'https://github.com/alleyinteractive/wordpress-simplechart/';
+
+	private $_latest_plugin_version;
+	private $_latest_plugin_zip_url;
+	private $_latest_plugin_update_name;
+	private $_latest_plugin_update_description;
+	private $_latest_plugin_last_updated;
+
+	public function __construct() {
+		add_action( 'admin_init', array( $this, 'simplechart_check_for_new_version' ) );
+		add_filter( 'plugins_api_result', array( $this, 'simplechart_mock_plugins_api' ), 10, 3 );
+	}
+
+	/**
+	 * Compares remote version transient to current version checking for update
+	 */
+	public function simplechart_check_for_new_version() {
+		if ( ! current_user_can( 'manage_options' ) ) {
+			return;
+		}
+
+		if ( false === ( $this->_latest_plugin_version = get_transient( 'simplechart_plugin_version_remote' ) ) ) {
+			$this->update_simplechart_remote_version();
+			return;
+		}
+
+		if ( false === ( $this->_latest_plugin_zip_url = get_transient( 'simplechart_plugin_zip_url_remote' ) ) ) {
+			$this->_latest_plugin_zip_url = $this->_simplechart_url;
+		}
+
+		if ( false === ( $this->_latest_plugin_update_name = get_transient( 'simplechart_plugin_update_name' ) ) ) {
+			$this->_latest_plugin_update_name = __( 'Latest Update', 'simplechart' );
+		}
+
+		if ( false === ( $this->_latest_plugin_update_description = get_transient( 'simplechart_plugin_update_body' ) ) ) {
+			$this->_latest_plugin_update_description = __( 'See more on Github.', 'simplechart' );
+		}
+
+		$this->_latest_plugin_last_updated = get_transient( 'simplechart_plugin_update_last_updated' );
+
+		if ( version_compare( Simplechart::instance()->get_config( 'version' ), $this->_latest_plugin_version, '<' ) ) {
+			add_filter( 'site_transient_update_plugins', array( $this, 'simplechart_extend_filter_update_plugins' ) );
+		}
+	}
+
+	/**
+	 * Add Simplechart to the plugin updates transient when appropraite
+	 * @param object          $update_plugins Contains plugin update info
```

Comment:
I typically see docblocks with an extra line here and single space after param type, like
```
/**
 * Add Simplechart to the plugin updates transient when appropraite
 * 
 * @param object $update_plugins Contains plugin update info
```
Also, `appropriate` 😜